modules/internal/OwnedObject.chpl

```diff
@@ -272,15 +251,10 @@ module OwnedObject {
        If this record was already managing a non-nil instance,
        that instance will be deleted.
      */
+    @deprecated(notes="owned.retain is deprecated - please use :proc:`owned.adopt` instead")
     proc ref retain(pragma "nil from arg" newPtr:unmanaged) {
-      if !isCoercible(newPtr.type, chpl_t) then
-        compilerError("cannot retain '" + newPtr.type:string + "' " +
-                      "(expected '" + _to_unmanaged(chpl_t):string + "')");
-
-      var oldPtr = chpl_p;
-      chpl_p = newPtr;
-      if oldPtr then
-        delete _to_unmanaged(oldPtr);
+      delete owned.release(this);
+      this = owned.adopt(newPtr);
```

Comment:
If `delete` is required here, it should be included in the deprecation note. Otherwise I'd rather remove it from this code.